src/server/kiwix-serve.cpp

```diff
@@ -453,6 +453,16 @@ static struct MHD_Response* handle_search(RequestContext* request_context)
       request_context->connection, MHD_GET_ARGUMENT_KIND, "pattern");
   std::string patternString
       = kiwix::urlDecode(pattern == NULL ? "" : string(pattern));
+
+  /* Check if we are on the welcome page or specific zim
+     Search from the welcome page should not be cached.
+  */
+  const char* contentArg = MHD_lookup_connection_value(
+      request_context->connection, MHD_GET_ARGUMENT_KIND, "content");
+  std::string contentArgString
+      = kiwix::urlDecode(contentArg == NULL ? "" : string(contentArg));
+  bool cacheEnabled = !contentArgString.empty();
+
```

Comment:
I think it would be simpler to test if `request_context->searcher == globalSearcher` or if `humanReadableBookId` is empty.

I think that specific handler should avoid to parse the the connection. Even if `handle_search` do it for the `pattern` it is better to avoid adding it if we can avoid it.